extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/HibernateValidatorRecorder.java

```diff
@@ -284,4 +285,10 @@ public ResteasyConfigSupport get() {
             }
         };
     }
+
+    // this is done in order to ensure that HibernateValidatorFactory is fully initialized at static init
+    // so completely in heap and ready to go when a native image is built
+    public void hibernateValidatorFactoryInit(BeanContainer beanContainer) {
+        HibernateValidatorFactory hibernateValidatorFactory = beanContainer.beanInstance(HibernateValidatorFactory.class);
```

Comment:
Note that this only works because `HibernateValidatorFactory` is `@Singleton` and not a normal scoped bean with a client proxy.

Reply:
Yup, I'm aware.  If it were ApplicationScoped I would have called toString